extensions/objects/templates/FilepathEditor.js

```diff
@@ -73,27 +162,321 @@ oppia.directive('filepathEditor', [
+              break;
+            case MOUSE_TOP:
+              adjustResizeTop(y);
+              break;
+            case MOUSE_TOP_RIGHT:
+              adjustResizeTop(y);
+              adjustResizeRight(x);
+              break;
+            case MOUSE_RIGHT:
+              adjustResizeRight(x);
+              break;
+            case MOUSE_BOTTOM_RIGHT:
+              adjustResizeBottom(y);
+              adjustResizeRight(x);
+              break;
+            case MOUSE_BOTTOM:
+              adjustResizeBottom(y);
+              break;
+            case MOUSE_BOTTOM_LEFT:
+              adjustResizeBottom(y);
+              adjustResizeLeft(x);
+              break;
+            case MOUSE_LEFT:
+              adjustResizeLeft(x);
+              break;
+          }
+        };
+
+        $scope.updatePositionWithinCroppingArea = function(x, y) {
+          var margin = CROP_BORDER_MARGIN_PX;
+          var cx1 = $scope.croppingArea.x1;
+          var cy1 = $scope.croppingArea.y1;
+          var cx2 = $scope.croppingArea.x2;
+          var cy2 = $scope.croppingArea.y2;
+
+          var xOnLeftBorder = x > cx1 - margin && x < cx1 + margin;
+          var xOnRightBorder = x > cx2 - margin && x < cx2 + margin;
+          var yOnTopBorder = y > cy1 - margin && y < cy1 + margin;
+          var yOnBottomBorder = y > cy2 - margin && y < cy2 + margin;
+          var xInside = x > cx1 && x < cx2;
+          var yInside = y > cy1 && y < cy2;
+          if (xOnLeftBorder && yOnTopBorder) {
+            // Upper left corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_TOP_LEFT;
+          } else if (xOnLeftBorder && yOnBottomBorder) {
+            // Lower left corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_BOTTOM_LEFT;
+          } else if (xOnLeftBorder) {
+            // Left border.
+            $scope.mousePositionWithinCroppingArea = MOUSE_LEFT;
+          } else if (xOnRightBorder && yOnTopBorder) {
+            // Upper right corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_TOP_RIGHT;
+          } else if (xOnRightBorder && yOnBottomBorder) {
+            // Lower right corner.
+            $scope.mousePositionWithinCroppingArea = MOUSE_BOTTOM_RIGHT;
+          } else if (xOnRightBorder) {
+            // Right border.
+            $scope.mousePositionWithinCroppingArea = MOUSE_RIGHT;
+          } else if (yOnTopBorder) {
+            // Top border.
+            $scope.mousePositionWithinCroppingArea = MOUSE_TOP;
+          } else if (yOnBottomBorder) {
+            // Bottom border.
+            $scope.mousePositionWithinCroppingArea = MOUSE_BOTTOM;
+          } else if (xInside && yInside) {
+            // Inside the cropping area.
+            $scope.mousePositionWithinCroppingArea = MOUSE_INSIDE;
+          } else {
+            $scope.mousePositionWithinCroppingArea = null;
+          }
+        };
+
+        $scope.mouseMoveOnImageArea = function(e) {
```

Comment:
Event handler names should start with on -- e.g. onMouseMoveOnImageArea(), or just onMouseMove() in this case.